query-engine/schema/src/output_types.rs

```diff
@@ -12,6 +12,23 @@ pub enum OutputType {
     Scalar(ScalarType),
 }
 
+impl std::fmt::Display for OutputType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Enum(_) => write!(f, "enum"),
+            Self::List(o) => write!(f, "{o}"),
+            Self::Object(o) => write!(
+                f,
+                "{}",
+                o.upgrade()
+                    .map(|f| f.identifier().name())
+                    .unwrap_or_else(|| String::from("Object"))
+            ),
+            Self::Scalar(s) => write!(f, "{s}"),
+        }
+    }
+}
```

Comment:
Could this be a function instead? Display impls have at least three disadvantages:

- They are global. Anywhere you put an output type in a format string, it will be used, and it may not be what you meant.
- They are not specific. This is for rendering error messages, but `impl Display` does not convey that.
- Display impls are used for different purposes in the same area of the codebase. In prisma-models, the Zipper Display impls are used for the type names rendered as part of DMMF, not for errors. I don't think the other Display impls are justified either, but this would add to the confusion.

Reply:
Yes, I can make that a pure function receiving an output type reference.